.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,38 @@
+on:
+    push:
+        branches:
+            - master
+    pull_request:
+        branches:
+            - "*"
+
+name: Build and Push Image
+
+jobs:
+    test:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+
+            - name: Login to Docker Hub
+              uses: docker/login-action@v3
+              with:
+                username: ${{ secrets.DOCKERHUB_USERNAME }}
+                password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+            - name: Make shared env
+              uses: ./.github/actions/make-shared-env
+
+            - name: Build app
+              env:
+                NODE_ENV: production
+              run: src/gradlew -p src :app:bootDistTar
+
+            - name: Build docker image
+              run: docker build -f ./Dockerfile --tag "${CI_ORG}/hint:${CI_SHA:0:7}" src
+
+            - name: Push image
```

Comment:
Are we able to use the docker/build-push-action here? I think there is support for tagging

Reply:
yep done